axelor-account/src/main/java/com/axelor/apps/account/web/InvoiceLineController.java

```diff
@@ -383,15 +395,182 @@ public void selectDefaultDistributionTemplate(ActionRequest request, ActionRespo
       throws AxelorException {
     try {
       InvoiceLine invoiceLine = request.getContext().asType(InvoiceLine.class);
-      InvoiceLineService invoiceLineService = Beans.get(InvoiceLineService.class);
-      invoiceLine = invoiceLineService.selectDefaultDistributionTemplate(invoiceLine);
+      InvoiceLineAnalyticService invoiceLineAnalyticService =
+          Beans.get(InvoiceLineAnalyticService.class);
+      invoiceLine = invoiceLineAnalyticService.selectDefaultDistributionTemplate(invoiceLine);
 
       response.setValue(
           "analyticDistributionTemplate", invoiceLine.getAnalyticDistributionTemplate());
       response.setValue(
           "analyticMoveLineList",
-          invoiceLineService.createAnalyticDistributionWithTemplate(invoiceLine));
+          invoiceLineAnalyticService.createAnalyticDistributionWithTemplate(invoiceLine));
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void createAnalyticAccountLines(ActionRequest request, ActionResponse response)
+      throws AxelorException {
```

Comment:
Please remove throw declaration when the method in fact does not throw the exception